Python Version Update for Docker

This is the error that I am getting when trying to bring in the code from PR #53 from the High Schoolers. Below is the error message:

(base) sarannagubandi@Sarans-MacBook-Pro SPAC_Shiny-Purdue-2025 % git branch
  Summer2025-SPAC-font_size_rk
  Summer2025-SPAC-heat_pin
  main
* merge/high-school-prs-issue-7
  test/bmc-paper-updates-pr-59
(base) sarannagubandi@Sarans-MacBook-Pro SPAC_Shiny-Purdue-2025 % make run
Building SPAC Shiny Docker image...
docker build -t spac-shiny .
[+] Building 72.1s (11/12)                                                                                                 docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                       0.0s
 => => transferring dockerfile: 1.14kB                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/python:3.9.19-slim-bookworm                                                             0.7s
 => [auth] library/python:pull token for registry-1.docker.io                                                                              0.0s
 => [internal] load .dockerignore                                                                                                          0.0s
 => => transferring context: 455B                                                                                                          0.0s
 => [1/7] FROM docker.io/library/python:3.9.19-slim-bookworm@sha256:69e712dbe4c4a166527cbf69374533125cfb6ee93a5e39031a0191c741d386d7       0.0s
 => => resolve docker.io/library/python:3.9.19-slim-bookworm@sha256:69e712dbe4c4a166527cbf69374533125cfb6ee93a5e39031a0191c741d386d7       0.0s
 => [internal] load build context                                                                                                          0.0s
 => => transferring context: 4.17kB                                                                                                        0.0s
 => CACHED [2/7] WORKDIR /app                                                                                                              0.0s
 => CACHED [3/7] RUN echo "Acquire::http::Pipeline-Depth 0;" > /etc/apt/apt.conf.d/99custom &&     echo "Acquire::http::No-Cache true;" >  0.0s
 => CACHED [4/7] RUN apt-get update && apt-get install -y     gcc     g++     git     libhdf5-dev     pkg-config     && rm -rf /var/lib/a  0.0s
 => CACHED [5/7] COPY requirements.txt .                                                                                                   0.0s
 => ERROR [6/7] RUN pip install --no-cache-dir -r requirements.txt                                                                        71.2s
------
 > [6/7] RUN pip install --no-cache-dir -r requirements.txt:
2.238 Collecting sag-py-execution-time-decorator@ git+https://github.com/SamhammerAG/sag_py_execution_time_decorator.git@976c9683d561aadc0166495117c42cc1762633d7
2.239   Cloning https://github.com/SamhammerAG/sag_py_execution_time_decorator.git (to revision 976c9683d561aadc0166495117c42cc1762633d7) to /tmp/pip-install-s7c7blpk/sag-py-execution-time-decorator_2a6f624fad0149a4a6f3353d44b4e123
2.244   Running command git clone --filter=blob:none --quiet https://github.com/SamhammerAG/sag_py_execution_time_decorator.git /tmp/pip-install-s7c7blpk/sag-py-execution-time-decorator_2a6f624fad0149a4a6f3353d44b4e123
2.989   Running command git rev-parse -q --verify 'sha^976c9683d561aadc0166495117c42cc1762633d7'
2.995   Running command git fetch -q https://github.com/SamhammerAG/sag_py_execution_time_decorator.git 976c9683d561aadc0166495117c42cc1762633d7
3.266   Running command git checkout -q 976c9683d561aadc0166495117c42cc1762633d7
3.567   Resolved https://github.com/SamhammerAG/sag_py_execution_time_decorator.git to commit 976c9683d561aadc0166495117c42cc1762633d7
3.581   Installing build dependencies: started
5.961   Installing build dependencies: finished with status 'done'
5.963   Getting requirements to build wheel: started
6.186   Getting requirements to build wheel: finished with status 'done'
6.188   Preparing metadata (pyproject.toml): started
6.406   Preparing metadata (pyproject.toml): finished with status 'done'
6.415 Collecting spac@ git+https://github.com/FNLCR-DMAP/SCSAWorkflow.git@49a6479d3083070064a6218f34125b5e86ad8e46#egg=spac
6.417   Cloning https://github.com/FNLCR-DMAP/SCSAWorkflow.git (to revision 49a6479d3083070064a6218f34125b5e86ad8e46) to /tmp/pip-install-s7c7blpk/spac_c5afff76922a41ea9861e1f2819921f2
6.423   Running command git clone --filter=blob:none --quiet https://github.com/FNLCR-DMAP/SCSAWorkflow.git /tmp/pip-install-s7c7blpk/spac_c5afff76922a41ea9861e1f2819921f2
7.770   Running command git rev-parse -q --verify 'sha^49a6479d3083070064a6218f34125b5e86ad8e46'
7.775   Running command git fetch -q https://github.com/FNLCR-DMAP/SCSAWorkflow.git 49a6479d3083070064a6218f34125b5e86ad8e46
8.095   Running command git checkout -q 49a6479d3083070064a6218f34125b5e86ad8e46
8.901   Resolved https://github.com/FNLCR-DMAP/SCSAWorkflow.git to commit 49a6479d3083070064a6218f34125b5e86ad8e46
8.908   Preparing metadata (setup.py): started
9.269   Preparing metadata (setup.py): finished with status 'done'
9.278 Collecting git+https://github.com/holoviz/datashader.git (from -r requirements.txt (line 58))
9.279   Cloning https://github.com/holoviz/datashader.git to /tmp/pip-req-build-k3vhhnh5
9.285   Running command git clone --filter=blob:none --quiet https://github.com/holoviz/datashader.git /tmp/pip-req-build-k3vhhnh5
14.11   Resolved https://github.com/holoviz/datashader.git to commit a5ae17c4a7b0e9b9e864c43da0ff9422dfe0c196
14.12   Installing build dependencies: started
17.32   Installing build dependencies: finished with status 'done'
17.32   Getting requirements to build wheel: started
17.42   Getting requirements to build wheel: finished with status 'done'
17.42   Preparing metadata (pyproject.toml): started
17.76   Preparing metadata (pyproject.toml): finished with status 'done'
18.41 Collecting numpy==1.26.4
18.50   Downloading numpy-1.26.4-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
19.57      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.2/18.2 MB 19.3 MB/s eta 0:00:00
19.97 Collecting pandas==1.5.3
20.02   Downloading pandas-1.5.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.2 MB)
20.60      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.2/12.2 MB 21.8 MB/s eta 0:00:00
20.70 Collecting anndata==0.10.9
20.72   Downloading anndata-0.10.9-py3-none-any.whl (128 kB)
20.72      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 129.0/129.0 kB 107.9 MB/s eta 0:00:00
20.81 Collecting scanpy==1.10.3
20.83   Downloading scanpy-1.10.3-py3-none-any.whl (2.1 MB)
20.95      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 18.4 MB/s eta 0:00:00
21.03 Collecting zarr==2.10.3
21.05   Downloading zarr-2.10.3-py3-none-any.whl (146 kB)
21.06      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 146.5/146.5 kB 17.4 MB/s eta 0:00:00
21.31 Collecting scikit-learn==1.1.1
21.33   Downloading scikit_learn-1.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (30.8 MB)
22.85      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 30.8/30.8 MB 23.2 MB/s eta 0:00:00
23.34 Collecting matplotlib==3.9.2
23.36   Downloading matplotlib-3.9.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.3 MB)
23.80      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.3/8.3 MB 19.3 MB/s eta 0:00:00
23.85 Collecting seaborn==0.13.2
23.87   Downloading seaborn-0.13.2-py3-none-any.whl (294 kB)
23.88      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 294.9/294.9 kB 30.0 MB/s eta 0:00:00
24.48 Collecting pillow==10.4.0
24.50   Downloading pillow-10.4.0-cp39-cp39-manylinux_2_28_x86_64.whl (4.5 MB)
24.74      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.5/4.5 MB 19.2 MB/s eta 0:00:00
24.79 Collecting squidpy==1.2.2
24.81   Downloading squidpy-1.2.2-py3-none-any.whl (179 kB)
24.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.8/179.8 kB 25.0 MB/s eta 0:00:00
25.00 Collecting scikit-image==0.19.3
25.02   Downloading scikit_image-0.19.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (14.0 MB)
25.85      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.0/14.0 MB 17.5 MB/s eta 0:00:00
26.28 Collecting scipy==1.10.1
26.30   Downloading scipy-1.10.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.5 MB)
28.07      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 34.5/34.5 MB 18.5 MB/s eta 0:00:00
28.21 Collecting ipykernel==6.29.5
28.23   Downloading ipykernel-6.29.5-py3-none-any.whl (117 kB)
28.24      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.2/117.2 kB 22.5 MB/s eta 0:00:00
28.34 Collecting ipython==8.18.0
28.36   Downloading ipython-8.18.0-py3-none-any.whl (808 kB)
28.40      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 808.2/808.2 kB 21.3 MB/s eta 0:00:00
28.43 Collecting ipython-genutils==0.2.0
28.44   Downloading ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
28.51 Collecting ipywidgets==8.1.7
28.55   Downloading ipywidgets-8.1.7-py3-none-any.whl (139 kB)
28.56      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 139.8/139.8 kB 38.3 MB/s eta 0:00:00
28.60 Collecting jsonschema-specifications==2023.12.1
28.62   Downloading jsonschema_specifications-2023.12.1-py3-none-any.whl (18 kB)
28.68 Collecting jupyter-events==0.10.0
28.70   Downloading jupyter_events-0.10.0-py3-none-any.whl (18 kB)
28.77 Collecting jupyter_client==7.4.9
28.79   Downloading jupyter_client-7.4.9-py3-none-any.whl (133 kB)
28.79      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.5/133.5 kB 70.6 MB/s eta 0:00:00
28.85 Collecting jupyter_core==5.7.2
28.87   Downloading jupyter_core-5.7.2-py3-none-any.whl (28 kB)
28.96 Collecting jupyter_server==2.14.1
28.98   Downloading jupyter_server-2.14.1-py3-none-any.whl (383 kB)
29.00      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 383.4/383.4 kB 22.9 MB/s eta 0:00:00
29.04 Collecting jupyter_server_terminals==0.5.3
29.05   Downloading jupyter_server_terminals-0.5.3-py3-none-any.whl (13 kB)
29.08 Collecting jupyterlab_pygments==0.3.0
29.10   Downloading jupyterlab_pygments-0.3.0-py3-none-any.whl (15 kB)
29.16 Collecting jupyterlab_widgets==3.0.15
29.18   Downloading jupyterlab_widgets-3.0.15-py3-none-any.whl (216 kB)
29.19      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 216.6/216.6 kB 39.1 MB/s eta 0:00:00
29.24 Collecting nbclient==0.10.0
29.26   Downloading nbclient-0.10.0-py3-none-any.whl (25 kB)
29.32 Collecting nbconvert==7.16.4
29.34   Downloading nbconvert-7.16.4-py3-none-any.whl (257 kB)
29.36      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 257.4/257.4 kB 36.1 MB/s eta 0:00:00
29.39 Collecting nbformat==5.10.4
29.41   Downloading nbformat-5.10.4-py3-none-any.whl (78 kB)
29.42      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.5/78.5 kB 34.6 MB/s eta 0:00:00
29.53 Collecting notebook==6.5.7
29.55   Downloading notebook-6.5.7-py3-none-any.whl (529 kB)
29.58      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 529.8/529.8 kB 19.2 MB/s eta 0:00:00
29.65 Collecting notebook_shim==0.2.4
29.67   Downloading notebook_shim-0.2.4-py3-none-any.whl (13 kB)
29.72 Collecting phenograph==1.5.7
29.74   Downloading PhenoGraph-1.5.7-py3-none-any.whl (159 kB)
29.75      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 159.6/159.6 kB 13.6 MB/s eta 0:00:00
29.94 Collecting plotly==5.22.0
29.96   Downloading plotly-5.22.0-py3-none-any.whl (16.4 MB)
30.91      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.4/16.4 MB 18.1 MB/s eta 0:00:00
31.11 Collecting python-dateutil==2.9.0.post0
31.13   Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
31.14      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 kB 39.6 MB/s eta 0:00:00
31.18 Collecting python-json-logger==2.0.7
31.20   Downloading python_json_logger-2.0.7-py3-none-any.whl (8.1 kB)
31.27 Collecting requests==2.32.3
31.29   Downloading requests-2.32.3-py3-none-any.whl (64 kB)
31.30      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 77.4 MB/s eta 0:00:00
32.18 Collecting rpds-py==0.19.0
32.20   Downloading rpds_py-0.19.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (356 kB)
32.23      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 356.0/356.0 kB 15.0 MB/s eta 0:00:00
32.25 Collecting rsconnect==0.1.3
32.27   Downloading rsconnect-0.1.3-py3-none-any.whl (2.3 kB)
32.32 Collecting rsconnect-jupyter==1.8.0
32.33   Downloading rsconnect_jupyter-1.8.0-py2.py3-none-any.whl (74 kB)
32.34      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.4/74.4 kB 229.0 MB/s eta 0:00:00
32.40 Collecting rsconnect_python==1.24.0
32.43   Downloading rsconnect_python-1.24.0-py2.py3-none-any.whl (122 kB)
32.44      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 122.9/122.9 kB 58.9 MB/s eta 0:00:00
32.49 Collecting shiny==0.10.2
32.52   Downloading shiny-0.10.2-py3-none-any.whl (3.4 MB)
32.72      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 16.9 MB/s eta 0:00:00
32.76 Collecting shinyswatch==0.6.1
32.78   Downloading shinyswatch-0.6.1-py3-none-any.whl (1.1 MB)
32.85      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 18.2 MB/s eta 0:00:00
32.88 Collecting shinywidgets==0.7.0
32.91   Downloading shinywidgets-0.7.0-py3-none-any.whl (1.8 MB)
33.00      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 20.9 MB/s eta 0:00:00
33.02 Collecting sinfo==0.3.4
33.05   Downloading sinfo-0.3.4.tar.gz (24 kB)
33.07   Preparing metadata (setup.py): started
33.26   Preparing metadata (setup.py): finished with status 'done'
33.34 Collecting tables==3.8.0
33.36   Downloading tables-3.8.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (6.5 MB)
33.65      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.5/6.5 MB 23.1 MB/s eta 0:00:00
33.69 Collecting tinycss2==1.3.0
33.71   Downloading tinycss2-1.3.0-py3-none-any.whl (22 kB)
33.80 Collecting tqdm==4.66.4
33.82   Downloading tqdm-4.66.4-py3-none-any.whl (78 kB)
33.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.3/78.3 kB 194.0 MB/s eta 0:00:00
33.88 Collecting typeguard==4.3.0
33.90   Downloading typeguard-4.3.0-py3-none-any.whl (35 kB)
33.94 Collecting types-python-dateutil==2.9.0.20240316
33.96   Downloading types_python_dateutil-2.9.0.20240316-py3-none-any.whl (9.7 kB)
34.06 Collecting typing_extensions==4.12.2
34.08   Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
34.13 Collecting websocket-client==1.8.0
34.15   Downloading websocket_client-1.8.0-py3-none-any.whl (58 kB)
34.16      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 17.5 MB/s eta 0:00:00
34.43 Collecting websockets==12.0
34.46   Downloading websockets-12.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (130 kB)
34.47      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 130.0/130.0 kB 119.7 MB/s eta 0:00:00
34.53 Collecting zipp==3.19.2
34.55   Downloading zipp-3.19.2-py3-none-any.whl (9.0 kB)
34.65 Collecting tifffile
34.67   Downloading tifffile-2024.8.30-py3-none-any.whl (227 kB)
34.68      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 227.3/227.3 kB 59.6 MB/s eta 0:00:00
34.96 Collecting shapely==2.0.7
34.98   Downloading shapely-2.0.7-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.5 MB)
35.09      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.5/2.5 MB 22.9 MB/s eta 0:00:00
35.14 Collecting colorcet
35.16   Downloading colorcet-3.1.0-py3-none-any.whl (260 kB)
35.18      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 260.3/260.3 kB 16.7 MB/s eta 0:00:00
35.27 Collecting pytz>=2020.1
35.29   Downloading pytz-2025.2-py2.py3-none-any.whl (509 kB)
35.31      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 509.2/509.2 kB 31.5 MB/s eta 0:00:00
35.48 Collecting h5py>=3.1
35.50   Downloading h5py-3.14.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.6 MB)
35.71      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.6/4.6 MB 22.3 MB/s eta 0:00:00
35.78 Collecting natsort
35.80   Downloading natsort-8.4.0-py3-none-any.whl (38 kB)
35.86 Collecting packaging>=20.0
35.88   Downloading packaging-25.0-py3-none-any.whl (66 kB)
35.88      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.5/66.5 kB 173.6 MB/s eta 0:00:00
35.92 Collecting array-api-compat!=1.5,>1.4
35.94   Downloading array_api_compat-1.11.2-py3-none-any.whl (53 kB)
35.94      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.1/53.1 kB 128.0 MB/s eta 0:00:00
35.98 Collecting exceptiongroup
36.00   Downloading exceptiongroup-1.3.0-py3-none-any.whl (16 kB)
36.10 Collecting get-annotations
36.13   Downloading get_annotations-0.1.2-py3-none-any.whl (4.5 kB)
36.26 Collecting statsmodels>=0.13
36.28   Downloading statsmodels-0.14.5-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (10.7 MB)
36.87      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.7/10.7 MB 18.1 MB/s eta 0:00:00
36.93 Collecting patsy
36.96   Downloading patsy-1.0.2-py2.py3-none-any.whl (233 kB)
36.97      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 233.3/233.3 kB 47.6 MB/s eta 0:00:00
37.02 Collecting joblib
37.05   Downloading joblib-1.5.2-py3-none-any.whl (308 kB)
37.06      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 308.4/308.4 kB 32.3 MB/s eta 0:00:00
37.12 Collecting networkx>=2.7
37.14   Downloading networkx-3.2.1-py3-none-any.whl (1.6 MB)
37.23      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 19.3 MB/s eta 0:00:00
37.32 Collecting umap-learn!=0.5.0,>=0.5
37.34   Downloading umap_learn-0.5.9.post2-py3-none-any.whl (90 kB)
37.35      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.1/90.1 kB 68.3 MB/s eta 0:00:00
37.39 Collecting pynndescent>=0.5
37.41   Downloading pynndescent-0.5.13-py3-none-any.whl (56 kB)
37.42      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.9/56.9 kB 73.6 MB/s eta 0:00:00
37.45 Collecting session-info
37.47   Downloading session_info-1.0.1-py3-none-any.whl (9.1 kB)
37.50 Collecting legacy-api-wrap>=1.4
37.52   Downloading legacy_api_wrap-1.4.1-py3-none-any.whl (10.0 kB)
37.90 Collecting numba>=0.56
37.92   Downloading numba-0.60.0-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (3.7 MB)
38.12      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 19.2 MB/s eta 0:00:00
38.25 Collecting numcodecs>=0.6.4
38.27   Downloading numcodecs-0.12.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.7 MB)
38.60      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.7/7.7 MB 23.6 MB/s eta 0:00:00
38.65 Collecting fasteners
38.66   Downloading fasteners-0.20-py3-none-any.whl (18 kB)
38.69 Collecting asciitree
38.72   Downloading asciitree-0.3.3.tar.gz (4.0 kB)
38.73   Preparing metadata (setup.py): started
38.94   Preparing metadata (setup.py): finished with status 'done'
39.02 Collecting threadpoolctl>=2.0.0
39.04   Downloading threadpoolctl-3.6.0-py3-none-any.whl (18 kB)
39.13 Collecting importlib-resources>=3.2.0
39.15   Downloading importlib_resources-6.5.2-py3-none-any.whl (37 kB)
39.34 Collecting kiwisolver>=1.3.1
39.35   Downloading kiwisolver-1.4.7-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.6 MB)
39.42      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 25.5 MB/s eta 0:00:00
39.63 Collecting contourpy>=1.0.1
39.65   Downloading contourpy-1.3.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (321 kB)
39.66      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 321.9/321.9 kB 57.6 MB/s eta 0:00:00
39.73 Collecting pyparsing>=2.3.1
39.75   Downloading pyparsing-3.2.5-py3-none-any.whl (113 kB)
39.76      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 113.9/113.9 kB 131.9 MB/s eta 0:00:00
40.26 Collecting fonttools>=4.22.0
40.27   Downloading fonttools-4.60.1-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (4.8 MB)
40.46      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.8/4.8 MB 25.8 MB/s eta 0:00:00
40.51 Collecting cycler>=0.10
40.53   Downloading cycler-0.12.1-py3-none-any.whl (8.3 kB)
40.66 Collecting dask-image>=0.5.0
40.68   Downloading dask_image-2024.5.3-py3-none-any.whl (59 kB)
40.69      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.5/59.5 kB 78.2 MB/s eta 0:00:00
40.81 Collecting dask[array]>=2021.02.0
40.83   Downloading dask-2024.8.0-py3-none-any.whl (1.2 MB)
40.87      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 31.3 MB/s eta 0:00:00
40.93 Collecting fsspec>=2021.11.0
40.95   Downloading fsspec-2025.10.0-py3-none-any.whl (200 kB)
40.96      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 201.0/201.0 kB 69.0 MB/s eta 0:00:00
42.67 Collecting aiohttp>=3.8.1
42.69   Downloading aiohttp-3.13.2-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (1.7 MB)
42.77      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 22.1 MB/s eta 0:00:00
42.82 Collecting validators>=0.18.2
42.84   Downloading validators-0.35.0-py3-none-any.whl (44 kB)
42.84      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.7/44.7 kB 87.5 MB/s eta 0:00:00
42.87 Collecting docrep>=0.3.1
42.88   Downloading docrep-0.3.2.tar.gz (33 kB)
42.91   Preparing metadata (setup.py): started
43.09   Preparing metadata (setup.py): finished with status 'done'
43.20 Collecting leidenalg>=0.8.2
43.22   Downloading leidenalg-0.10.2-cp38-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)
43.30      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.0/2.0 MB 27.2 MB/s eta 0:00:00
43.34 Collecting matplotlib-scalebar>=0.8.0
43.36   Downloading matplotlib_scalebar-0.9.0-py3-none-any.whl (16 kB)
43.39 Collecting omnipath>=1.0.5
43.42   Downloading omnipath-1.0.12-py3-none-any.whl (51 kB)
43.43      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 51.6/51.6 kB 112.7 MB/s eta 0:00:00
43.52 Collecting xarray>=0.16.1
43.54   Downloading xarray-2024.7.0-py3-none-any.whl (1.2 MB)
43.60      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 22.7 MB/s eta 0:00:00
43.73 Collecting imageio>=2.4.1
43.75   Downloading imageio-2.37.0-py3-none-any.whl (315 kB)
43.76      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 315.8/315.8 kB 33.5 MB/s eta 0:00:00
43.88 Collecting PyWavelets>=1.1.1
43.90   Downloading pywavelets-1.6.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.5 MB)
44.13      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.5/4.5 MB 20.7 MB/s eta 0:00:00
44.34 Collecting tornado>=6.1
44.36   Downloading tornado-6.5.2-cp39-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)
44.38      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 443.9/443.9 kB 35.4 MB/s eta 0:00:00
44.41 Collecting matplotlib-inline>=0.1
44.43   Downloading matplotlib_inline-0.2.1-py3-none-any.whl (9.5 kB)
44.50 Collecting traitlets>=5.4.0
44.52   Downloading traitlets-5.14.3-py3-none-any.whl (85 kB)
44.52      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.4/85.4 kB 255.4 MB/s eta 0:00:00
44.73 Collecting psutil
44.75   Downloading psutil-7.1.2-cp36-abi3-manylinux2010_x86_64.manylinux_2_12_x86_64.manylinux_2_28_x86_64.whl (258 kB)
44.77      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 258.7/258.7 kB 43.1 MB/s eta 0:00:00
44.82 Collecting comm>=0.1.1
44.84   Downloading comm-0.2.3-py3-none-any.whl (7.3 kB)
44.88 Collecting nest-asyncio
44.90   Downloading nest_asyncio-1.6.0-py3-none-any.whl (5.2 kB)
45.81 Collecting pyzmq>=24
45.83   Downloading pyzmq-27.1.0-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (863 kB)
45.89      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 863.8/863.8 kB 15.1 MB/s eta 0:00:00
46.20 Collecting debugpy>=1.6.5
46.23   Downloading debugpy-1.8.17-cp39-cp39-manylinux_2_34_x86_64.whl (3.1 MB)
46.36      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 23.9 MB/s eta 0:00:00
46.48 Collecting pygments>=2.4.0
46.50   Downloading pygments-2.19.2-py3-none-any.whl (1.2 MB)
46.56      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 24.0 MB/s eta 0:00:00
46.60 Collecting jedi>=0.16
46.62   Downloading jedi-0.19.2-py2.py3-none-any.whl (1.6 MB)
46.70      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 21.5 MB/s eta 0:00:00
46.75 Collecting pexpect>4.3
46.77   Downloading pexpect-4.9.0-py2.py3-none-any.whl (63 kB)
46.77      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.8/63.8 kB 54.6 MB/s eta 0:00:00
46.80 Collecting stack-data
46.81   Downloading stack_data-0.6.3-py3-none-any.whl (24 kB)
46.86 Collecting decorator
46.88   Downloading decorator-5.2.1-py3-none-any.whl (9.2 kB)
46.96 Collecting prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30
46.98   Downloading prompt_toolkit-3.0.52-py3-none-any.whl (391 kB)
46.99      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 391.4/391.4 kB 37.2 MB/s eta 0:00:00
47.08 Collecting widgetsnbextension~=4.0.14
47.10   Downloading widgetsnbextension-4.0.14-py3-none-any.whl (2.2 MB)
47.19      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.2/2.2 MB 24.7 MB/s eta 0:00:00
47.31 Collecting referencing>=0.31.0
47.33   Downloading referencing-0.36.2-py3-none-any.whl (26 kB)
47.46 Collecting pyyaml>=5.3
47.49   Downloading pyyaml-6.0.3-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (750 kB)
47.52      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 750.8/750.8 kB 33.0 MB/s eta 0:00:00
47.54 Collecting rfc3986-validator>=0.1.1
47.56   Downloading rfc3986_validator-0.1.1-py2.py3-none-any.whl (4.2 kB)
47.61 Collecting rfc3339-validator
47.62   Downloading rfc3339_validator-0.1.4-py2.py3-none-any.whl (3.5 kB)
47.69 Collecting jsonschema[format-nongpl]>=4.18.0
47.71   Downloading jsonschema-4.25.1-py3-none-any.whl (90 kB)
47.71      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.0/90.0 kB 58.4 MB/s eta 0:00:00
47.79 Collecting entrypoints
47.81   Downloading entrypoints-0.4-py3-none-any.whl (5.3 kB)
47.89 Collecting platformdirs>=2.5
47.91   Downloading platformdirs-4.4.0-py3-none-any.whl (18 kB)
47.99 Collecting jinja2>=3.0.3
48.00   Downloading jinja2-3.1.6-py3-none-any.whl (134 kB)
48.01      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 134.9/134.9 kB 40.4 MB/s eta 0:00:00
48.06 Collecting anyio>=3.1.0
48.08   Downloading anyio-4.11.0-py3-none-any.whl (109 kB)
48.09      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 109.1/109.1 kB 63.6 MB/s eta 0:00:00
48.12 Collecting send2trash>=1.8.2
48.14   Downloading Send2Trash-1.8.3-py3-none-any.whl (18 kB)
48.19 Collecting prometheus-client>=0.9
48.21   Downloading prometheus_client-0.23.1-py3-none-any.whl (61 kB)
48.21      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.1/61.1 kB 148.0 MB/s eta 0:00:00
48.29 Collecting argon2-cffi>=21.1
48.30   Downloading argon2_cffi-25.1.0-py3-none-any.whl (14 kB)
48.34 Collecting terminado>=0.8.3
48.36   Downloading terminado-0.18.1-py3-none-any.whl (14 kB)
48.39 Collecting overrides>=5.0
48.41   Downloading overrides-7.7.0-py3-none-any.whl (17 kB)
48.70 Collecting markupsafe>=2.0
48.72   Downloading markupsafe-3.0.3-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (20 kB)
48.75 Collecting pandocfilters>=1.4.1
48.77   Downloading pandocfilters-1.5.1-py2.py3-none-any.whl (8.7 kB)
48.87 Collecting importlib-metadata>=3.6
48.89   Downloading importlib_metadata-8.7.0-py3-none-any.whl (27 kB)
48.94 Collecting mistune<4,>=2.0.3
48.96   Downloading mistune-3.1.4-py3-none-any.whl (53 kB)
48.96      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.5/53.5 kB 129.8 MB/s eta 0:00:00
49.02 Collecting beautifulsoup4
49.03   Downloading beautifulsoup4-4.14.2-py3-none-any.whl (106 kB)
49.04      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.4/106.4 kB 75.6 MB/s eta 0:00:00
49.09 Collecting bleach!=5.0.0
49.10   Downloading bleach-6.2.0-py3-none-any.whl (163 kB)
49.11      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.4/163.4 kB 84.1 MB/s eta 0:00:00
49.14 Collecting defusedxml
49.16   Downloading defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
49.21 Collecting fastjsonschema>=2.15
49.23   Downloading fastjsonschema-2.21.2-py3-none-any.whl (24 kB)
49.37 Collecting nbclassic>=0.4.7
49.39   Downloading nbclassic-1.3.3-py3-none-any.whl (11.5 MB)
50.04      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.5/11.5 MB 17.4 MB/s eta 0:00:00
50.14 Requirement already satisfied: setuptools>=18.0.1 in /usr/local/lib/python3.9/site-packages (from phenograph==1.5.7->-r requirements.txt (line 30)) (58.1.0)
50.20 Collecting tenacity>=6.2.0
50.22   Downloading tenacity-9.1.2-py3-none-any.whl (28 kB)
50.26 Collecting six>=1.5
50.28   Downloading six-1.17.0-py2.py3-none-any.whl (11 kB)
50.38 Collecting urllib3<3,>=1.21.1
50.40   Downloading urllib3-2.5.0-py3-none-any.whl (129 kB)
50.40      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 129.8/129.8 kB 32.3 MB/s eta 0:00:00
50.45 Collecting certifi>=2017.4.17
50.48   Downloading certifi-2025.10.5-py3-none-any.whl (163 kB)
50.49      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.3/163.3 kB 55.7 MB/s eta 0:00:00
50.74 Collecting charset-normalizer<4,>=2
50.76   Downloading charset_normalizer-3.4.4-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (153 kB)
50.76      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.0/154.0 kB 40.8 MB/s eta 0:00:00
50.80 Collecting idna<4,>=2.5
50.82   Downloading idna-3.11-py3-none-any.whl (71 kB)
50.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.0/71.0 kB 180.6 MB/s eta 0:00:00
51.00 Collecting pins
51.02   Downloading pins-0.9.1-py2.py3-none-any.whl (122 kB)
51.03      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 122.1/122.1 kB 33.2 MB/s eta 0:00:00
51.14 Collecting semver<3.0.0,>=2.0.0
51.17   Downloading semver-2.13.0-py2.py3-none-any.whl (12 kB)
51.22 Collecting click>=8.0.0
51.24   Downloading click-8.1.8-py3-none-any.whl (98 kB)
51.24      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.2/98.2 kB 197.3 MB/s eta 0:00:00
51.29 Collecting pyjwt>=2.4.0
51.30   Downloading PyJWT-2.10.1-py3-none-any.whl (22 kB)
51.31 Requirement already satisfied: pip>=10.0.0 in /usr/local/lib/python3.9/site-packages (from rsconnect_python==1.24.0->-r requirements.txt (line 38)) (23.0.1)
51.45 Collecting starlette
51.48   Downloading starlette-0.49.1-py3-none-any.whl (74 kB)
51.48      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.2/74.2 kB 57.0 MB/s eta 0:00:00
51.53 Collecting htmltools>=0.5.1
51.55   Downloading htmltools-0.6.0-py3-none-any.whl (84 kB)
51.56      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.0/85.0 kB 56.2 MB/s eta 0:00:00
51.61 Collecting asgiref>=3.5.2
51.63   Downloading asgiref-3.10.0-py3-none-any.whl (24 kB)
51.67 Collecting linkify-it-py>=1.0
51.69   Downloading linkify_it_py-2.0.3-py3-none-any.whl (19 kB)
51.75 Collecting uvicorn>=0.16.0
51.77   Downloading uvicorn-0.38.0-py3-none-any.whl (68 kB)
51.78      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 68.1/68.1 kB 164.3 MB/s eta 0:00:00
51.82 Collecting python-multipart>=0.0.7
51.83   Downloading python_multipart-0.0.20-py3-none-any.whl (24 kB)
51.88 Collecting mdit-py-plugins>=0.3.0
51.90   Downloading mdit_py_plugins-0.4.2-py3-none-any.whl (55 kB)
51.90      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.3/55.3 kB 100.9 MB/s eta 0:00:00
51.93 Collecting appdirs>=1.4.4
51.95   Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
52.01 Collecting markdown-it-py>=1.1.0
52.03   Downloading markdown_it_py-3.0.0-py3-none-any.whl (87 kB)
52.03      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 87.5/87.5 kB 180.7 MB/s eta 0:00:00
52.07 Collecting questionary>=2.0.0
52.09   Downloading questionary-2.1.1-py3-none-any.whl (36 kB)
52.35 Collecting watchfiles>=0.18.0
52.37   Downloading watchfiles-1.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (457 kB)
52.39      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 457.5/457.5 kB 22.5 MB/s eta 0:00:00
52.51 Collecting anywidget
52.53   Downloading anywidget-0.9.18-py3-none-any.whl (220 kB)
52.54      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 220.7/220.7 kB 22.2 MB/s eta 0:00:00
52.60 Collecting stdlib_list
52.62   Downloading stdlib_list-0.12.0-py3-none-any.whl (87 kB)
52.64      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 87.6/87.6 kB 3.7 MB/s eta 0:00:00
52.86 Collecting numexpr>=2.6.2
52.88   Downloading numexpr-2.10.2-cp39-cp39-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl (396 kB)
52.91      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 396.7/396.7 kB 17.1 MB/s eta 0:00:00
53.61 Collecting cython>=0.29.21
53.64   Downloading cython-3.1.6-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (3.4 MB)
53.80      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 21.5 MB/s eta 0:00:00
53.84 Collecting py-cpuinfo
53.86   Downloading py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)
54.16 Collecting blosc2~=2.0.0
54.18   Downloading blosc2-2.0.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.9 MB)
54.38      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 20.2 MB/s eta 0:00:00
54.42 Collecting webencodings>=0.4
54.44   Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
54.72 Collecting pyct
54.74   Downloading pyct-0.6.0-py3-none-any.whl (16 kB)
54.77 Collecting toolz
54.79   Downloading toolz-1.1.0-py3-none-any.whl (58 kB)
54.80      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.1/58.1 kB 69.0 MB/s eta 0:00:00
54.83 Collecting multipledispatch
54.85   Downloading multipledispatch-1.0.0-py3-none-any.whl (12 kB)
54.92 Collecting param
54.95   Downloading param-2.2.1-py3-none-any.whl (119 kB)
54.95      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 119.0/119.0 kB 55.7 MB/s eta 0:00:00
54.96 INFO: pip is looking at multiple versions of colorcet to determine which version is compatible with other requirements. This could take a while.
54.97 Collecting colorcet
54.99   Downloading colorcet-3.0.1-py2.py3-none-any.whl (1.7 MB)
55.08      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 20.3 MB/s eta 0:00:00
55.18   Downloading colorcet-3.0.0-py2.py3-none-any.whl (1.6 MB)
55.25      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 26.2 MB/s eta 0:00:00
55.35   Downloading colorcet-2.0.6-py2.py3-none-any.whl (1.6 MB)
55.42      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 26.8 MB/s eta 0:00:00
55.50   Downloading colorcet-2.0.5-py2.py3-none-any.whl (1.6 MB)
55.60      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 18.3 MB/s eta 0:00:00
55.70   Downloading colorcet-2.0.2-py2.py3-none-any.whl (1.6 MB)
55.78      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 21.0 MB/s eta 0:00:00
55.88   Downloading colorcet-2.0.1-py2.py3-none-any.whl (1.6 MB)
55.95      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 21.8 MB/s eta 0:00:00
56.04   Downloading colorcet-2.0.0-py2.py3-none-any.whl (1.6 MB)
56.14      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 17.0 MB/s eta 0:00:00
56.20 INFO: pip is looking at multiple versions of colorcet to determine which version is compatible with other requirements. This could take a while.
56.23   Downloading colorcet-1.0.1-py2.py3-none-any.whl (1.4 MB)
56.31      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.4/1.4 MB 18.5 MB/s eta 0:00:00
56.39   Downloading colorcet-0.9.1.tar.gz (115 kB)
56.40      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.1/115.1 kB 93.2 MB/s eta 0:00:00
56.42   Preparing metadata (setup.py): started
56.59   Preparing metadata (setup.py): finished with status 'done'
56.65   Downloading colorcet-0.9.0.tar.gz (115 kB)
56.66      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.1/115.1 kB 131.9 MB/s eta 0:00:00
56.67   Preparing metadata (setup.py): started
56.84   Preparing metadata (setup.py): finished with status 'done'
56.87 INFO: pip is looking at multiple versions of spac to determine which version is compatible with other requirements. This could take a while.
56.87 INFO: pip is looking at multiple versions of sag-py-execution-time-decorator to determine which version is compatible with other requirements. This could take a while.
56.87 INFO: pip is looking at multiple versions of tifffile to determine which version is compatible with other requirements. This could take a while.
56.88 Collecting tifffile
56.91   Downloading tifffile-2024.8.28-py3-none-any.whl (226 kB)
56.92      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.1/226.1 kB 73.4 MB/s eta 0:00:00
57.01   Downloading tifffile-2024.8.24-py3-none-any.whl (225 kB)
57.02      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.9/225.9 kB 93.5 MB/s eta 0:00:00
57.12   Downloading tifffile-2024.8.10-py3-none-any.whl (225 kB)
57.13      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.8/225.8 kB 56.4 MB/s eta 0:00:00
57.23   Downloading tifffile-2024.7.24-py3-none-any.whl (226 kB)
57.24      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.2/226.2 kB 40.4 MB/s eta 0:00:00
57.32   Downloading tifffile-2024.7.21-py3-none-any.whl (225 kB)
57.33      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.0/226.0 kB 88.4 MB/s eta 0:00:00
57.41   Downloading tifffile-2024.7.2-py3-none-any.whl (225 kB)
57.44      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.9/225.9 kB 11.4 MB/s eta 0:00:00
57.52   Downloading tifffile-2024.6.18-py3-none-any.whl (226 kB)
57.54      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 226.0/226.0 kB 63.2 MB/s eta 0:00:00
57.59 INFO: pip is looking at multiple versions of sag-py-execution-time-decorator to determine which version is compatible with other requirements. This could take a while.
57.59 INFO: pip is looking at multiple versions of tifffile to determine which version is compatible with other requirements. This could take a while.
57.62   Downloading tifffile-2024.5.22-py3-none-any.whl (225 kB)
57.63      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.5/225.5 kB 28.5 MB/s eta 0:00:00
57.71   Downloading tifffile-2024.5.10-py3-none-any.whl (225 kB)
57.72      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.7/225.7 kB 33.1 MB/s eta 0:00:00
57.80   Downloading tifffile-2024.5.3-py3-none-any.whl (225 kB)
57.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.3/225.3 kB 12.6 MB/s eta 0:00:00
57.90   Downloading tifffile-2024.4.24-py3-none-any.whl (225 kB)
57.91      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.0/225.0 kB 22.9 MB/s eta 0:00:00
57.99   Downloading tifffile-2024.4.18-py3-none-any.whl (224 kB)
58.00      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 225.0/225.0 kB 38.1 MB/s eta 0:00:00
58.05 INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
58.05 INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
58.08   Downloading tifffile-2024.2.12-py3-none-any.whl (224 kB)
58.09      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 224.5/224.5 kB 32.1 MB/s eta 0:00:00
58.18   Downloading tifffile-2024.1.30-py3-none-any.whl (224 kB)
58.19      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 224.1/224.1 kB 126.9 MB/s eta 0:00:00
58.28   Downloading tifffile-2023.12.9-py3-none-any.whl (223 kB)
58.29      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 223.6/223.6 kB 104.1 MB/s eta 0:00:00
58.38   Downloading tifffile-2023.9.26-py3-none-any.whl (222 kB)
58.39      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 222.9/222.9 kB 83.1 MB/s eta 0:00:00
58.47   Downloading tifffile-2023.9.18-py3-none-any.whl (222 kB)
58.48      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 222.1/222.1 kB 53.7 MB/s eta 0:00:00
58.55   Downloading tifffile-2023.8.30-py3-none-any.whl (221 kB)
58.56      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 221.9/221.9 kB 70.9 MB/s eta 0:00:00
58.64   Downloading tifffile-2023.8.25-py3-none-any.whl (221 kB)
58.66      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 221.7/221.7 kB 127.6 MB/s eta 0:00:00
58.74   Downloading tifffile-2023.8.12-py3-none-any.whl (220 kB)
58.75      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 221.0/221.0 kB 69.9 MB/s eta 0:00:00
58.83   Downloading tifffile-2023.7.18-py3-none-any.whl (221 kB)
58.85      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 221.4/221.4 kB 21.0 MB/s eta 0:00:00
58.93   Downloading tifffile-2023.7.10-py3-none-any.whl (220 kB)
58.95      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 220.9/220.9 kB 15.1 MB/s eta 0:00:00
59.03   Downloading tifffile-2023.7.4-py3-none-any.whl (220 kB)
59.06      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 220.8/220.8 kB 7.5 MB/s eta 0:00:00
59.16   Downloading tifffile-2023.4.12-py3-none-any.whl (219 kB)
59.17      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 219.4/219.4 kB 41.1 MB/s eta 0:00:00
59.26   Downloading tifffile-2023.3.21-py3-none-any.whl (218 kB)
59.27      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 218.7/218.7 kB 64.3 MB/s eta 0:00:00
59.36   Downloading tifffile-2023.3.15-py3-none-any.whl (218 kB)
59.37      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 218.6/218.6 kB 57.6 MB/s eta 0:00:00
59.46   Downloading tifffile-2023.2.28-py3-none-any.whl (216 kB)
59.47      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 216.4/216.4 kB 71.6 MB/s eta 0:00:00
59.56   Downloading tifffile-2023.2.27-py3-none-any.whl (216 kB)
59.57      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 216.1/216.1 kB 50.8 MB/s eta 0:00:00
59.65   Downloading tifffile-2023.2.3-py3-none-any.whl (215 kB)
59.66      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 215.1/215.1 kB 52.9 MB/s eta 0:00:00
59.75   Downloading tifffile-2023.2.2-py3-none-any.whl (214 kB)
59.76      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 215.0/215.0 kB 61.2 MB/s eta 0:00:00
59.89   Downloading tifffile-2023.1.23.1-py3-none-any.whl (214 kB)
59.90      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.8/214.8 kB 23.3 MB/s eta 0:00:00
59.99   Downloading tifffile-2023.1.23-py3-none-any.whl (214 kB)
60.00      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 214.8/214.8 kB 98.3 MB/s eta 0:00:00
60.08   Downloading tifffile-2022.10.10-py3-none-any.whl (210 kB)
60.09      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 210.3/210.3 kB 29.9 MB/s eta 0:00:00
60.18   Downloading tifffile-2022.8.12-py3-none-any.whl (208 kB)
60.19      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 208.5/208.5 kB 42.5 MB/s eta 0:00:00
60.27   Downloading tifffile-2022.8.8-py3-none-any.whl (208 kB)
60.28      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 208.1/208.1 kB 34.8 MB/s eta 0:00:00
60.35   Downloading tifffile-2022.8.3-py3-none-any.whl (207 kB)
60.36      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 207.9/207.9 kB 128.1 MB/s eta 0:00:00
60.44   Downloading tifffile-2022.7.31-py3-none-any.whl (207 kB)
60.45      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 207.4/207.4 kB 78.0 MB/s eta 0:00:00
60.52   Downloading tifffile-2022.7.28-py3-none-any.whl (207 kB)
60.54      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 207.0/207.0 kB 26.2 MB/s eta 0:00:00
60.61   Downloading tifffile-2022.5.4-py3-none-any.whl (195 kB)
60.62      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 195.6/195.6 kB 28.1 MB/s eta 0:00:00
60.70   Downloading tifffile-2022.4.28-py3-none-any.whl (193 kB)
60.71      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 193.9/193.9 kB 84.2 MB/s eta 0:00:00
60.78   Downloading tifffile-2022.4.26-py3-none-any.whl (191 kB)
60.80      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 191.8/191.8 kB 14.2 MB/s eta 0:00:00
60.87   Downloading tifffile-2022.4.8-py3-none-any.whl (180 kB)
60.88      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 180.6/180.6 kB 48.3 MB/s eta 0:00:00
60.96   Downloading tifffile-2022.3.25-py3-none-any.whl (179 kB)
60.97      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.9/179.9 kB 49.9 MB/s eta 0:00:00
61.05   Downloading tifffile-2022.3.16-py3-none-any.whl (179 kB)
61.06      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.2/179.2 kB 69.0 MB/s eta 0:00:00
61.13   Downloading tifffile-2022.2.9-py3-none-any.whl (180 kB)
61.14      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 180.3/180.3 kB 120.9 MB/s eta 0:00:00
61.22   Downloading tifffile-2022.2.2-py3-none-any.whl (179 kB)
61.24      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.9/179.9 kB 58.1 MB/s eta 0:00:00
61.34   Downloading tifffile-2021.11.2-py3-none-any.whl (178 kB)
61.36      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 178.9/178.9 kB 65.0 MB/s eta 0:00:00
61.46   Downloading tifffile-2021.10.12-py3-none-any.whl (175 kB)
61.47      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 175.0/175.0 kB 54.7 MB/s eta 0:00:00
61.55   Downloading tifffile-2021.10.10-py3-none-any.whl (174 kB)
61.56      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 174.9/174.9 kB 43.3 MB/s eta 0:00:00
61.64   Downloading tifffile-2021.8.30-py3-none-any.whl (171 kB)
61.65      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 171.9/171.9 kB 39.1 MB/s eta 0:00:00
61.72   Downloading tifffile-2021.8.8-py3-none-any.whl (171 kB)
61.73      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 171.8/171.8 kB 54.0 MB/s eta 0:00:00
61.81   Downloading tifffile-2021.7.30-py3-none-any.whl (171 kB)
61.82      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 171.4/171.4 kB 29.1 MB/s eta 0:00:00
61.89   Downloading tifffile-2021.7.2-py3-none-any.whl (169 kB)
61.89      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 169.9/169.9 kB 84.4 MB/s eta 0:00:00
61.97   Downloading tifffile-2021.6.14-py3-none-any.whl (169 kB)
61.99      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 169.1/169.1 kB 52.6 MB/s eta 0:00:00
62.07   Downloading tifffile-2021.6.6-py3-none-any.whl (168 kB)
62.08      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 168.5/168.5 kB 18.9 MB/s eta 0:00:00
62.16   Downloading tifffile-2021.4.8-py3-none-any.whl (165 kB)
62.17      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 165.0/165.0 kB 18.4 MB/s eta 0:00:00
62.25   Downloading tifffile-2021.3.31-py3-none-any.whl (165 kB)
62.26      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 165.1/165.1 kB 165.5 MB/s eta 0:00:00
62.33   Downloading tifffile-2021.3.17-py3-none-any.whl (163 kB)
62.34      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 163.5/163.5 kB 162.6 MB/s eta 0:00:00
62.42   Downloading tifffile-2021.3.16-py3-none-any.whl (164 kB)
62.43      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 164.6/164.6 kB 149.0 MB/s eta 0:00:00
62.50   Downloading tifffile-2021.3.5-py3-none-any.whl (162 kB)
62.51      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 162.1/162.1 kB 99.7 MB/s eta 0:00:00
62.59   Downloading tifffile-2021.3.4-py3-none-any.whl (161 kB)
62.59      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 161.8/161.8 kB 93.0 MB/s eta 0:00:00
62.67   Downloading tifffile-2021.2.26-py3-none-any.whl (160 kB)
62.68      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 160.9/160.9 kB 70.1 MB/s eta 0:00:00
62.76   Downloading tifffile-2021.2.1-py3-none-any.whl (159 kB)
62.76      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 159.7/159.7 kB 43.5 MB/s eta 0:00:00
62.84   Downloading tifffile-2021.1.14-py3-none-any.whl (158 kB)
62.85      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.9/158.9 kB 170.0 MB/s eta 0:00:00
62.93   Downloading tifffile-2021.1.11-py3-none-any.whl (158 kB)
62.94      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.4/158.4 kB 18.8 MB/s eta 0:00:00
63.02   Downloading tifffile-2021.1.8-py3-none-any.whl (158 kB)
63.02      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.3/158.3 kB 79.5 MB/s eta 0:00:00
63.10   Downloading tifffile-2020.12.8-py3-none-any.whl (157 kB)
63.11      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 157.7/157.7 kB 51.2 MB/s eta 0:00:00
63.19   Downloading tifffile-2020.12.4-py3-none-any.whl (157 kB)
63.20      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 157.4/157.4 kB 45.4 MB/s eta 0:00:00
63.28   Downloading tifffile-2020.11.26-py3-none-any.whl (157 kB)
63.28      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 157.0/157.0 kB 42.9 MB/s eta 0:00:00
63.36   Downloading tifffile-2020.11.18-py3-none-any.whl (153 kB)
63.37      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 153.9/153.9 kB 63.4 MB/s eta 0:00:00
63.46   Downloading tifffile-2020.10.1-py3-none-any.whl (152 kB)
63.47      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 152.6/152.6 kB 68.0 MB/s eta 0:00:00
63.54   Downloading tifffile-2020.9.30-py3-none-any.whl (152 kB)
63.55      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 152.5/152.5 kB 29.7 MB/s eta 0:00:00
63.63   Downloading tifffile-2020.9.29-py3-none-any.whl (154 kB)
63.64      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.6/154.6 kB 25.5 MB/s eta 0:00:00
63.71   Downloading tifffile-2020.9.28-py3-none-any.whl (154 kB)
63.72      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.5/154.5 kB 50.0 MB/s eta 0:00:00
63.80   Downloading tifffile-2020.9.22-py3-none-any.whl (153 kB)
63.81      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 153.7/153.7 kB 52.9 MB/s eta 0:00:00
63.89   Downloading tifffile-2020.9.3-py3-none-any.whl (148 kB)
63.90      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 148.4/148.4 kB 38.1 MB/s eta 0:00:00
63.97   Downloading tifffile-2020.8.25-py3-none-any.whl (147 kB)
63.98      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 147.1/147.1 kB 22.9 MB/s eta 0:00:00
64.07   Downloading tifffile-2020.8.13-py3-none-any.whl (146 kB)
64.08      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 146.6/146.6 kB 104.8 MB/s eta 0:00:00
64.17   Downloading tifffile-2020.7.24-py3-none-any.whl (146 kB)
64.21      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 146.2/146.2 kB 4.4 MB/s eta 0:00:00
64.36   Downloading tifffile-2020.7.22-py3-none-any.whl (145 kB)
64.37      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 145.4/145.4 kB 26.1 MB/s eta 0:00:00
64.48   Downloading tifffile-2020.7.17-py3-none-any.whl (144 kB)
64.54      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 144.0/144.0 kB 2.6 MB/s eta 0:00:00
64.63   Downloading tifffile-2020.7.4-py3-none-any.whl (138 kB)
64.65      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 138.9/138.9 kB 63.3 MB/s eta 0:00:00
64.74   Downloading tifffile-2020.6.3-py3-none-any.whl (133 kB)
64.75      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.2/133.2 kB 135.1 MB/s eta 0:00:00
64.86   Downloading tifffile-2020.5.30-py3-none-any.whl (133 kB)
64.87      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 kB 19.4 MB/s eta 0:00:00
64.95   Downloading tifffile-2020.5.25-py3-none-any.whl (132 kB)
64.96      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.8/132.8 kB 24.8 MB/s eta 0:00:00
65.09   Downloading tifffile-2020.5.11-py3-none-any.whl (132 kB)
65.10      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.4/132.4 kB 63.1 MB/s eta 0:00:00
65.28 Collecting imagecodecs>=2020.2.18
65.30   Downloading imagecodecs-2024.12.30-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (43.5 MB)
68.59      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 43.5/43.5 MB 9.8 MB/s eta 0:00:00
68.69 Collecting tifffile
68.71   Downloading tifffile-2020.5.7-py3-none-any.whl (136 kB)
68.72      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 136.2/136.2 kB 12.9 MB/s eta 0:00:00
68.80   Downloading tifffile-2020.5.5-py3-none-any.whl (132 kB)
68.80      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.2/132.2 kB 146.3 MB/s eta 0:00:00
68.89   Downloading tifffile-2020.2.16-py3-none-any.whl (130 kB)
68.89      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 130.1/130.1 kB 61.8 MB/s eta 0:00:00
68.99   Downloading tifffile-2019.7.26.2-py2.py3-none-any.whl (131 kB)
69.00      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 131.4/131.4 kB 69.2 MB/s eta 0:00:00
69.23 Collecting imagecodecs-lite>=2019.12.2
69.26   Downloading imagecodecs_lite-2022.9.26-py3-none-any.whl (1.4 kB)
69.31 Collecting tifffile
69.35   Downloading tifffile-2019.7.26-py2.py3-none-any.whl (131 kB)
69.36      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 131.3/131.3 kB 69.2 MB/s eta 0:00:00
69.43 INFO: pip is looking at multiple versions of shapely to determine which version is compatible with other requirements. This could take a while.
69.43 INFO: pip is looking at multiple versions of zipp to determine which version is compatible with other requirements. This could take a while.
69.43 INFO: pip is looking at multiple versions of websockets to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of websocket-client to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of typing-extensions to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of types-python-dateutil to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of typeguard to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of tqdm to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of tinycss2 to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of tables to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of sinfo to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of shinywidgets to determine which version is compatible with other requirements. This could take a while.
69.44 INFO: pip is looking at multiple versions of shinyswatch to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of shiny to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of rsconnect-python to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of rsconnect-jupyter to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of rsconnect to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of rpds-py to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of python-json-logger to determine which version is compatible with other requirements. This could take a while.
69.45 INFO: pip is looking at multiple versions of python-dateutil to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of plotly to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of phenograph to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of notebook-shim to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of notebook to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of nbformat to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of nbconvert to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of nbclient to determine which version is compatible with other requirements. This could take a while.
69.46 INFO: pip is looking at multiple versions of jupyterlab-widgets to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyterlab-pygments to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyter-server-terminals to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyter-server to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyter-core to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyter-client to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jupyter-events to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of jsonschema-specifications to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of ipywidgets to determine which version is compatible with other requirements. This could take a while.
69.47 INFO: pip is looking at multiple versions of ipython-genutils to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of ipython to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of ipykernel to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of scipy to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of scikit-image to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of squidpy to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of pillow to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of seaborn to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of matplotlib to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of scikit-learn to determine which version is compatible with other requirements. This could take a while.
69.48 INFO: pip is looking at multiple versions of zarr to determine which version is compatible with other requirements. This could take a while.
69.49 INFO: pip is looking at multiple versions of scanpy to determine which version is compatible with other requirements. This could take a while.
69.49 INFO: pip is looking at multiple versions of anndata to determine which version is compatible with other requirements. This could take a while.
69.49 INFO: pip is looking at multiple versions of pandas to determine which version is compatible with other requirements. This could take a while.
69.49 INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
69.49 INFO: pip is looking at multiple versions of numpy to determine which version is compatible with other requirements. This could take a while.
69.49 ERROR: Package 'datashader' requires a different Python: 3.9.19 not in '>=3.10'
69.74 
69.74 [notice] A new release of pip is available: 23.0.1 -> 25.3
69.74 [notice] To update, run: pip install --upgrade pip
------
Dockerfile:28
--------------------
  26 |     
  27 |     # Install Python dependencies
  28 | >>> RUN pip install --no-cache-dir -r requirements.txt
  29 |     
  30 |     # Copy the application code
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c pip install --no-cache-dir -r requirements.txt" did not complete successfully: exit code: 1
make: *** [build] Error 1
(base) sarannagubandi@Sarans-MacBook-Pro SPAC_Shiny-Purdue-2025 %

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,14 +1,10 @@
 # Use official Python image as base
-# Fix: python version conflicts (Issue #15 in Purdue-2025 fork)
-# Error message when building docker:
-# Package 'datashader' requires a different Python: 3.9.19 not in '>=3.10'
-# FROM python:3.9.19-slim-bookworm
-FROM python:3.10-slim-bookworm
+FROM python:3.9.19-slim-bookworm
 
 # Set working directory
 WORKDIR /app
 
-# Fix: 'Hash Sum Mismatch' bug for mac device (Issue #12 in Purdue-2025 fork)
+# Fix 'Hash Sum Mismatch' bug for mac device
 RUN echo "Acquire::http::Pipeline-Depth 0;" > /etc/apt/apt.conf.d/99custom && \
     echo "Acquire::http::No-Cache true;" >> /etc/apt/apt.conf.d/99custom && \
     echo "Acquire::BrokenProxy    true;" >> /etc/apt/apt.conf.d/99custom

--- a/environment.yml
+++ b/environment.yml
@@ -25,7 +25,7 @@ channels:
   - leej3
   - https://fnlcr-dmap.github.io/scimap/
 dependencies:
-  - python=3.10.19  # To support datashader, need python>=3.10
+  - python=3.9.19   # To be consistent with requirement.txt
   - numpy=1.26.4
   - pandas=1.5.3
   - anndata=0.10.9
@@ -38,6 +38,10 @@ dependencies:
   - squidpy=1.2.2
   - scikit-image=0.19.3
   - scipy=1.10.1
+  # Added: packages for heatmap mode in scatterplot
+  - colorcet
+  - datashader==0.16.3
+  - dask==2024.2.1    # Needed older version of 'dask' to be compatible with panda 1.5.3
   # For macOS (MacArm64) users, uncomment the following three lines
   # - tables>=3.8.0
   # - c-blosc2
@@ -86,9 +90,6 @@ dependencies:
       - tifffile
       - shapely==2.0.7  # To be consistent with requirement.txt
       - sag-py-execution-time-decorator @ git+https://github.com/SamhammerAG/sag_py_execution_time_decorator.git@976c9683d561aadc0166495117c42cc1762633d7
-      # Added: packages for heatmap mode in scatterplot
-      - colorcet
-      - git+https://github.com/holoviz/datashader.git
       # SPAC package pinned to specific commit for template compatibility (dev branch as of 2025-10-15)
       - spac @ git+https://github.com/FNLCR-DMAP/SCSAWorkflow.git@c2a248826b91880c62308e85e60cae9361c275d0#egg=spac
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -54,6 +54,7 @@ shapely==2.0.7
 sag-py-execution-time-decorator @ git+https://github.com/SamhammerAG/sag_py_execution_time_decorator.git@976c9683d561aadc0166495117c42cc1762633d7
 # Added: packages for heatmap mode in scatterplot
 colorcet
-git+https://github.com/holoviz/datashader.git
+datashader==0.16.3
+dask==2024.2.1
 # SPAC package pinned to specific commit for template compatibility (dev branch as of 2025-10-15)
 spac @ git+https://github.com/FNLCR-DMAP/SCSAWorkflow.git@c2a248826b91880c62308e85e60cae9361c275d0#egg=spac

